app/assets/javascripts/discourse/app/services/notifications.js

```diff
@@ -0,0 +1,35 @@
+import Service, { service } from "@ember/service";
+import { disableImplicitInjections } from "discourse/lib/implicit-injections";
+import ReactiveTargetDate from "discourse/lib/reactive-target-date";
+
+@disableImplicitInjections
+export default class NotificationsService extends Service {
+  @service currentUser;
+
+  #dndReactiveTargetDate;
+
+  /**
+   * Whether the current user is in do not disturb mode.
+   * This getter is autotrackable, and will recompute when the 'do not disturb until'
+   * value changes, or when the threshold is passed.
+   */
+  get isInDoNotDisturb() {
+    if (!this.currentUser) {
+      return false;
+    }
+
+    if (!this.currentUser.do_not_disturb_until) {
+      return false;
+    }
+
+    this.#dndReactiveTargetDate ??= new ReactiveTargetDate(
+      () => this.currentUser.do_not_disturb_until
+    );
+
+    return !this.#dndReactiveTargetDate.hasPassed;
+  }
+
+  willDestroy() {
+    this.#dndReactiveTargetDate?.destroy();
```

Comment:
wouldn't it be better to use the getOwner/setOwner pattern so that it gets destroyed automatically?

you can forget calling the destructor but you can't forget passing in a "required"/first argument 😛

Reply:
That was my first try. But unfortunately getOwner/setOwner is separate from Ember's "destroyable" system. Destroying the application does not automatically destroy things which are owned by it. 😬